File: 0169. Majority Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        index = 0
        count = 1
        
        for i in range(1,len(nums)):
            if nums[index] == nums[i]:
                count += 1
            else:
                count -= 1
                if count == 0 :
                    index = i
                    count = 1
                    
        
        return nums[index]

File: 0169. Majority Element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        var index = 0;
        var count = 1;
        
        for(var i = 1 ; i<nums.Length ; i++){
            if (nums[index] == nums[i]){
                count ++;
            }else{
                count --;
                if (count == 0){
                    index = i;
                    count = 1;
                }
                    
            }
        }
                
        
        return nums[index];
    }
}